File: PHP/8/example8.php
#!/usr/bin/env php

<?php

include_once('people.php');

foreach ($people as $p)
{
    echo "Person ...\n";

    foreach ($p as $k => $v)
    {
        echo "$k: " . $v . "\n";

        if ($k === "age")
        {
            $years_left = 65 - $v;
        }
    }

    echo "Years left: $years_left\n";

    echo "\n";

}

?>

File: Perl/8/example8.pl
#!/usr/bin/env perl

use strict;
use warnings;

require "people.pl";

our @people;

foreach my $person (@people)
{

    print "Person ...\n";
    my $years_left = 0;

    foreach (keys %{$person})
    {
        print "$_: ${$person}{$_}\n";

        if ($_ eq "age")
        {
            $years_left = 65 - ${$person}{$_} or die("Cannot convert <<age>> to integer.\n");
        }
    }

    printf("Years left: %d\n", $years_left);
    print "\n";
}